truffle/src/main/java/org/jruby/truffle/nodes/core/TrufflePrimitiveNodes.java

```diff
@@ -565,4 +575,93 @@ public DynamicObject printInterleavedBacktrace() {
+
+        public SpawnProcess(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        @Specialization(guards = {
+                "isRubyString(command)",
+                "isRubyArray(arguments)",
+                "isRubyArray(environmentVariables)" })
+        public int spawn(DynamicObject command,
+                         DynamicObject arguments,
+                         DynamicObject environmentVariables) {
+
+            final long longPid = call(
+                    StringOperations.getString(command),
+                    toStringArray(arguments),
+                    toStringArray(environmentVariables));
+            assert longPid <= Integer.MAX_VALUE;
+            // VMWaitPidPrimitiveNode accepts only int
+            final int pid = (int) longPid;
+
+            if (pid == -1) {
+                // TODO (pitr 07-Sep-2015): needs compatibility improvements
+                throw new RaiseException(getContext().getCoreLibrary().errnoError(getContext().getPosix().errno(), this));
+            }
+
+            return pid;
+        }
+
+        private String[] toStringArray(DynamicObject rubyStrings) {
+            final int size = Layouts.ARRAY.getSize(rubyStrings);
+            final Object[] unconvertedStrings = ArrayOperations.toObjectArray(rubyStrings);
+            final String[] strings = new String[size];
+
+            for (int i = 0; i < size; i++) {
+                assert Layouts.STRING.isString(unconvertedStrings[i]);
+                strings[i] = StringOperations.getString((DynamicObject) unconvertedStrings[i]);
+            }
+
+            return strings;
+        }
+
+        @TruffleBoundary
+        private long call(String command, String[] arguments, String[] environmentVariables) {
+            // TODO (pitr 04-Sep-2015): only simple implementation, does not support file actions or other options
+            return getContext().getPosix().posix_spawnp(
+                    command,
+                    Collections.<SpawnFileAction>emptyList(),
+                    Arrays.asList(arguments),
+                    Arrays.asList(environmentVariables));
+
+        }
+    }
+
+    @CoreMethod(names = "set_global_status_variable", onSingleton = true, required = 1)
+    @NodeChild(value = "value", type = RubyNode.class)
+    public abstract static class SetGlobalStatusVariable extends CoreMethodNode {
+
+        @Child private WriteInstanceVariableNode writeNode;
+
+        public SetGlobalStatusVariable(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        abstract RubyNode getValue();
+
+        @ShortCircuit("value")
+        protected boolean needsValue() {
+            return false;
+        }
+
+        @Specialization
+        public DynamicObject set(VirtualFrame frame, boolean hasValue, DynamicObject value) {
+            if (writeNode == null) {
```

Comment:
Why lazy? Eager sounds better, and a normal RubyNode instead of CoreMethodNode.
If lazy makes sense, please use `@Cached`.